ui/src/app/tags/components/DefinitionField/DefinitionField.tsx

```diff
@@ -33,10 +33,14 @@ const getDefinitionError = (
 };
 
 export const DefinitionField = (): JSX.Element => {
-  const { errors } = useFormikContext<CreateParams>();
+  const { initialValues, errors, values } = useFormikContext<
+    CreateParams | UpdateParams
+  >();
   const definitionErrorId = useId();
   const definitionError = getDefinitionError(errors, definitionErrorId);
-
+  const hasChangedDefinition =
+    !!initialValues.definition &&
```

Comment:
I'm not sure about using the form initial values here since it might not necessarily be up to date. Its existence is determined by a value on the tag itself, so I think it would be better handled using that value directly. e.g.:
``` ts
const showDefinitionWarning = !!tag.definition && values.definition !== tag.definition;
```

Reply:
Done